src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -2813,6 +2813,54 @@ public void Test_Issue5099()
             result.Template!.Should().HaveValueAtPath("$.outputs.productGroupsResourceIds.metadata.description", "The Resources Ids of the API management service product groups");
         }
 
+        [TestMethod]
+        // https://github.com/Azure/bicep/issues/5371
+        public void Test_Issue5371_positive_test()
+        {
+            var result = CompilationHelper.Compile(@"
+var myValue = -9223372036854775808
+");
+
+            result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics();
+        }
+
+        [TestMethod]
+        // https://github.com/Azure/bicep/issues/5371
+        public void Test_Issue5371_positive_test_2()
+        {
+            var result = CompilationHelper.Compile(@"
+var myValue = 9223372036854775807
+");
+
+            result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics();
+        }
```

Comment:
We could combine the two tests above with [`DataRow`](https://www.meziantou.net/mstest-v2-data-tests.htm#using-datarow). E.g.:

```
[DataTestMethod]
[DataRow("var myValue = -922337203685477580")]
[DataRow("var myValue = 9223372036854775807")]
public void Test_Issue5371_positive(string fileContents)
{
  ...
}
```

Reply:
The same goes for the negative tests below.

Reply:
Spoke offline with Anthony and decided to add validation that the template is formatted correctly and includes the right integer literal or function expression such as json('-9223372036854775808'). This would mean we can't use DataRow for the positive tests since they would have different values.

However, the DataRow method can still be used for the negative tests and I have added the changes!

Reply:
I wonder if that's still possible because `DataRow` can take multiple parameters:
```C#
[DataRow("var myValue = -922337203685477580", "...json('-922337203685477580')...")]
public void Test_Issue5371_positive(string bicepFileContents, string expectedTemplateContents)
```